packages/webapp/src/components/ui/dialog.tsx

```diff
@@ -0,0 +1,133 @@
+"use client"
+
+import * as React from "react"
+import * as DialogPrimitive from "@radix-ui/react-dialog"
+import { X } from "lucide-react"
+
+import { cn } from "src/utils/ui-utils"
+
+const Dialog = DialogPrimitive.Root
+
+const DialogTrigger = DialogPrimitive.Trigger
+
+const DialogPortal = DialogPrimitive.Portal
+
+const DialogClose = DialogPrimitive.Close
+
+// ref: https://ui.shadcn.com/docs/components/dialog
+const DialogOverlay = React.forwardRef<
+  React.ElementRef<typeof DialogPrimitive.Overlay>,
+  React.ComponentPropsWithoutRef<typeof DialogPrimitive.Overlay>
+>(({ className, ...props }, ref) => (
+  <DialogPrimitive.Overlay
+    ref={ref}
+    className={cn(
+      "fixed inset-0 z-50 bg-black/80  data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0",
+      className
+    )}
+    {...props}
+  />
+))
+DialogOverlay.displayName = DialogPrimitive.Overlay.displayName
+
+
+interface CustomDialogContentProps extends React.ComponentPropsWithoutRef<typeof DialogPrimitive.Content> {
+  // extend the existing type definition to allow for an additional prop
+  canCloseExternally?: boolean;
+}
+
+const DialogContent = React.forwardRef<
+  React.ElementRef<typeof DialogPrimitive.Content>,
+  CustomDialogContentProps
+>(({ className, children, canCloseExternally = true, ...props }, ref) => (
```

Comment:
Okay to use a single property instead of closeable / surroundCloseable.

The point of that distinction is that you want to have some modals that the user can close but shouldn't close carelessly by clicking outside, like some error displays. I don't think we were using that capability anywhere, so a single property will do for now. Also we can always make modal closeable with an explicit "okay" button which is probably better anyway.